File: foodbridge-backend/src/models/FoodRequest.js
import mongoose from "mongoose";

const foodRequestSchema = new mongoose.Schema(
  {
    foodId: { type: mongoose.Schema.Types.ObjectId, ref: "Donation", required: true, index: true },
    donorId: { type: mongoose.Schema.Types.ObjectId, ref: "User", required: true, index: true },
    receiverId: { type: mongoose.Schema.Types.ObjectId, ref: "User", required: true, index: true },
    foodName: { type: String, required: true },
    foodType: { type: String, required: true },
    donorOrg: { type: String, required: true },
    receiverName: { type: String, required: true },
    receiverOrg: { type: String, required: true },
    receiverType: { type: String, required: true },
    message: { type: String, trim: true, maxlength: 300, default: "" },
    status: { type: String, enum: ["pending", "approved", "rejected", "collected"], default: "pending", index: true },
  },
  { timestamps: true }
);

foodRequestSchema.index({ foodId: 1, receiverId: 1 }, { unique: true });

export default mongoose.model("FoodRequest", foodRequestSchema);

File: foodbridge-backend/src/main/java/com/foodbridge/entity/FoodRequest.java
package com.foodbridge.entity;

import java.time.Instant;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.mongodb.core.index.CompoundIndex;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "foodRequests")
@CompoundIndex(name = "food_receiver_unique", def = "{'foodId': 1, 'receiverId': 1}", unique = true)
public class FoodRequest {

    @Id
    private String id;

    @Indexed
    private String foodId;

    @Indexed
    private String donorId;

    @Indexed
    private String receiverId;

    private String foodName;
    private String foodType;
    private String donorOrg;
    private String receiverName;
    private String receiverOrg;
    private String receiverType;
    private String message;

    @Indexed
    @Builder.Default
    private String status = "pending";

    @CreatedDate
    private Instant createdAt;

    @LastModifiedDate
    private Instant updatedAt;
}
